lib/tasks/whitehall_importer.rb

```diff
@@ -66,12 +66,16 @@ def most_recent_edition
     end
 
     def create_or_update_document
-      Document.find_or_create_by(
+      first_edition = whitehall_document["editions"].first
+      first_author = first_edition["history"].select { |h| h["event"] == "create" }.first
+
+      Document.find_or_create_by!(
         content_id: whitehall_document["content_id"],
         locale: "en",
         document_type_id: "news_story", ## To be updated once Whitehall exports this value
         created_at: whitehall_document["created_at"],
         updated_at: whitehall_document["updated_at"],
+        created_by_id: user_map.select { |u| u["whitehall_user_id"] == first_author["whodunnit"] }.first["content_publisher_user_id"],
```

Comment:
This would benefit from being extracted to a method, in the next two commits too:

```ruby
created_by_id: whitehall_uid_to_signon_uid(first_author["whodunnit"])
```

Reply:
Just to expand on this if you used a hash of whitehall user ids to content publisher user ids this could be down to:`user_ids[first_author["whodunnit"]`